File: Microsoft .NET/ASP.NET/C# Web Services/ASP.NET Web Services/ReportingServiceTypes.cs
namespace combit.Services
{
    public sealed class Error
    {
        internal Error()
        {
            Code = 0;
        }

        internal Error(string newDescription)
            : this(-1, newDescription)
        {
        }

        internal Error(int newCode, string newDescription)
        {
            Code = newCode;
            Description = newDescription;
        }

        public int Code { get; set; }
        public string Description { get; set; }
        public bool Succeeded { get { return (Code == 0); } }
    }

    public sealed class Response<T>
    {
        internal Response()
        {
            Status = new Error();
        }

        internal Response(T value)
        {
            Status = new Error();
            Value = value;
        }

        internal Response(Error status)
        {
            Status = status;
        }

        public Error Status { get; set; }
        public T Value { get; set; }
        public string RequesterID { get; set; }
    }

    public sealed class Response
    {
        internal Response()
        {
            Status = new Error();
        }

        internal Response(Error status)
        {
            Status = status;
        }

        public Error Status { get; set; }
    }

    public sealed class DataSource
    {
        internal DataSource(string newId, string newName, string newDescripton, string newImage)
        {
            Id = newId;
            Name = newName;
            Description = newDescripton;
            Image = newImage;
        }

        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
    }

    public sealed class Report
    {
        internal Report(string newDatasource, string newName, string newDescripton, string newImage)
        {
            Datasource = newDatasource;
            Name = newName;
            Description = newDescripton;
            Image = newImage;
        }

        public string Datasource { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
    }

    public class ReportFormat
    {
        internal ReportFormat(string newId, string newDescripton, string newImage)
        {
            Id = newId;
            Description = newDescripton;
            Image = newImage;
        }

        public string Id { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
    }
}

File: Microsoft .NET/.NET Framework 4/ASP.NET/VB.NET/Web Services Sample/ASP.NET Web Services/App_Code/ReportingServiceTypes.vb
Namespace combit.Services
    Public NotInheritable Class [Error]
        Friend Sub New()
            Code = 0
        End Sub

        Friend Sub New(newDescription As String)
            Me.New(-1, newDescription)
        End Sub

        Friend Sub New(newCode As Integer, newDescription As String)
            Code = newCode
            Description = newDescription
        End Sub

        Public Property Code() As Integer
            Get
                Return m_Code
            End Get
            Set(value As Integer)
                m_Code = value
            End Set
        End Property
        Private m_Code As Integer
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(value As String)
                m_Description = value
            End Set
        End Property
        Private m_Description As String
        Public ReadOnly Property Succeeded() As Boolean
            Get
                Return (Code = 0)
            End Get
        End Property
    End Class

    Public NotInheritable Class Response(Of T)
        Friend Sub New()
            Status = New [Error]()
        End Sub

        Friend Sub New(value__1 As T)
            Status = New [Error]()
            Value = value__1
        End Sub

        Friend Sub New(status__1 As [Error])
            Status = status__1
        End Sub

        Public Property Status() As [Error]
            Get
                Return m_Status
            End Get
            Set(value As [Error])
                m_Status = value
            End Set
        End Property
        Private m_Status As [Error]
        Public Property Value() As T
            Get
                Return m_Value
            End Get
            Set(value As T)
                m_Value = value
            End Set
        End Property
        Private m_Value As T
        Public Property RequesterID() As String
            Get
                Return m_RequesterID
            End Get
            Set(value As String)
                m_RequesterID = value
            End Set
        End Property
        Private m_RequesterID As String
    End Class

    Public NotInheritable Class Response
        Friend Sub New()
            Status = New [Error]()
        End Sub

        Friend Sub New(status__1 As [Error])
            Status = status__1
        End Sub

        Public Property Status() As [Error]
            Get
                Return m_Status
            End Get
            Set(value As [Error])
                m_Status = value
            End Set
        End Property
        Private m_Status As [Error]
    End Class

    Public NotInheritable Class DataSource
        Friend Sub New(newId As String, newName As String, newDescripton As String, newImage As String)
            Id = newId
            Name = newName
            Description = newDescripton
            Image = newImage
        End Sub

        Public Property Id() As String
            Get
                Return m_Id
            End Get
            Set(value As String)
                m_Id = value
            End Set
        End Property
        Private m_Id As String
        Public Property Name() As String
            Get
                Return m_Name
            End Get
            Set(value As String)
                m_Name = value
            End Set
        End Property
        Private m_Name As String
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(value As String)
                m_Description = value
            End Set
        End Property
        Private m_Description As String
        Public Property Image() As String
            Get
                Return m_Image
            End Get
            Set(value As String)
                m_Image = value
            End Set
        End Property
        Private m_Image As String
    End Class

    Public NotInheritable Class Report
        Friend Sub New(newDatasource As String, newName As String, newDescripton As String, newImage As String)
            Datasource = newDatasource
            Name = newName
            Description = newDescripton
            Image = newImage
        End Sub

        Public Property Datasource() As String
            Get
                Return m_Datasource
            End Get
            Set(value As String)
                m_Datasource = value
            End Set
        End Property
        Private m_Datasource As String
        Public Property Name() As String
            Get
                Return m_Name
            End Get
            Set(value As String)
                m_Name = value
            End Set
        End Property
        Private m_Name As String
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(value As String)
                m_Description = value
            End Set
        End Property
        Private m_Description As String
        Public Property Image() As String
            Get
                Return m_Image
            End Get
            Set(value As String)
                m_Image = value
            End Set
        End Property
        Private m_Image As String
    End Class

    Public Class ReportFormat
        Friend Sub New(newId As String, newDescripton As String, newImage As String)
            Id = newId
            Description = newDescripton
            Image = newImage
        End Sub

        Public Property Id() As String
            Get
                Return m_Id
            End Get
            Set(value As String)
                m_Id = value
            End Set
        End Property
        Private m_Id As String
        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(value As String)
                m_Description = value
            End Set
        End Property
        Private m_Description As String
        Public Property Image() As String
            Get
                Return m_Image
            End Get
            Set(value As String)
                m_Image = value
            End Set
        End Property
        Private m_Image As String
    End Class
End Namespace
